src/search/documents/user.py

```diff
@@ -73,25 +83,80 @@ def prepare_author_profile(self, instance) -> dict[str, Any] | None:
 
     # Used specifically for "autocomplete" style suggest feature
     def prepare_full_name_suggest(self, instance) -> dict[str, Any]:
-        full_name_suggest = ""
-        try:
-            full_name_suggest = (
-                f"{instance.author_profile.first_name} "
-                f"{instance.author_profile.last_name}"
-            )
-        except Exception:
-            # Some legacy users don't have an author profile
-            full_name_suggest = f"{instance.first_name} {instance.last_name}"
+        MAX_INPUTS = 10
+        VERIFIED_USER_WEIGHT_BONUS = 500
 
-        weight = instance.reputation
-
-        if instance.is_verified:
-            weight += 500
-
-        return {
-            "input": full_name_suggest.split() + [full_name_suggest],
-            "weight": weight,
-        }
+        # Get names with fallback to user model
+        try:
+            first = (instance.author_profile.first_name or "").strip()
+            last = (instance.author_profile.last_name or "").strip()
+        except AttributeError:
+            first = (instance.first_name or "").strip()
+            last = (instance.last_name or "").strip()
+
+        full_name = f"{first} {last}".strip()
+        if not full_name:
+            return {"input": [], "weight": 0}
+
+        # Build input list with priority order
+        inputs, seen = [], set()
+
+        def add_unique(value):
+            if value and value not in seen:
+                inputs.append(value)
+                seen.add(value)
+
+        # Add core names and combinations
```

Comment:
I think a comment that describes what these permutations may look like is useful here:

e.g., “José López”, “jose lopez”, “François”, “francois”